### PR TITLE
build: updating to support trunk based development

### PR DESCRIPTION
**Pull request by drduhe**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,7 +14,5 @@ jobs:
         with:
           node-version: '18.x'
       - run: npm ci
-      - run: npm run build
+      - run: npm run build --if-present
       - uses: pre-commit/action@v3.0.0
-      - name: Jest coverage Report
-        uses: ArtiomTr/jest-coverage-report-action@v2.3.0
```
Comment (RanbirAulakh) [inline .github/workflows/build.yml:19]: Please re-add this back in. Useful for test coverage.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -8,30 +8,24 @@ jobs:
     if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
     runs-on: ubuntu-latest
     steps:
-      - name: 'Checkout source code'
-        uses: actions/checkout@v4
+      - uses: actions/checkout@v4
       - name: 'cat package.json'
         run: cat ./package.json
-      - name: 'Automated Version Bump'
+      - uses: 'phips28/gh-action-bump-version@master'
```
Comment (RanbirAulakh) [inline .github/workflows/publish.yml:14]: Remove. Would prefer manual release for now.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    uses: ./.github/workflows/npm-release.yml
+    secrets: inherit
+  docs:
+    needs: [build]
+    uses: ./.github/workflows/docs.yml
+    secrets: inherit
+  notify:
+    needs: [build, docs]
+    uses: ./.github/workflows/notify.yml
+    secrets: inherit
```
Comment (RanbirAulakh) [inline .github/workflows/release.yml:22]: Remove this block of code. Not needed

File: CONTRIBUTING.md
Comment (RanbirAulakh) [inline CONTRIBUTING.md:1]: Remove `images` folder as well

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  build:
+    uses: ./.github/workflows/build.yml
+    secrets: inherit
+  publish:
+    needs: [build]
+    uses: ./.github/workflows/npm-release.yml
+    secrets: inherit
+  docs:
+    needs: [build]
```
Comment (RanbirAulakh) [inline .github/workflows/release.yml:16]: Update the needs to `publish`. Would prefer

build -> publish -> docs

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  push:
+    branches: ["main", "dev"]
+
+jobs:
+  build:
+    uses: ./.github/workflows/build.yml
+    secrets: inherit
+  publish:
+    needs: [build]
+    uses: ./.github/workflows/npm-release.yml
```
Comment (RanbirAulakh) [inline .github/workflows/release.yml:13]: Update the workflow path, its `publish.yml`

**Review by RanbirAulakh — APPROVED** [review]

**Review by jtblack-aws — APPROVED** [review]

**Pull request merged**